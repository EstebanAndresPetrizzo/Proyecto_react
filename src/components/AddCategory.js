import React, { useState } from 'react'
import PropTypes from 'prop-types';

export const AddCategory = ({setCategories}) => {
    const [inputValue, setInputValue] = useState('');
    const handleinputChange = (e)=>{
        setInputValue(e.target.value);
    }

    const handleSubmit = (e)=>{
        e.preventDefault();//Evita que se refresque la pagina
        if(inputValue.trim().length > 2){
            setCategories(cats => [inputValue,...cats]);
        }
    }
    return (
        <form onSubmit= {handleSubmit}>
            <p>{inputValue}</p>
            <input type="text"
            value={inputValue}
            onChange={handleinputChange}
            />
        </form>
    )
}

AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
}
