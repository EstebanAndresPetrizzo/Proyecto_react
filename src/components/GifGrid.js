import React/*, { useState, useEffect } */from 'react'
import PropTypes from 'prop-types'

import {useFechGifs} from '../hooks/useFechGifs'
import { GifGridItem } from './GifGridItem';
//import { getGifs } from './helpers/getGifs';

export const GifGrid = ({ category }) => {

    const {data:images,loading} = useFechGifs(category)
/*
    const [images, setImages] = useState([]);
    useEffect(() => {
        getGifs(category)
            .then(setImages);
    }, [category])//el segundo argumento son las dependencias, si se deja vac√≠o [], entonces se ejecutara una sola vez
*/
    return (
        <>
            <h3 className="animate__animated animate__zoomIn">{category}</h3>
            {loading && <p className="animate__animated animate__flash">Loading...</p>}
            <div className="card-grid">
                {
                    images.map(img => (
                        <GifGridItem
                            key={img.id}
                            {...img}//envia cada una de la propiedades
                        />
                    ))
                }
            </div>
        </>
    )
}

GifGrid.protoType = {
    category: PropTypes.string.isRequired
}