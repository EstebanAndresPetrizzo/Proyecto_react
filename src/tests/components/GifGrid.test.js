import { shallow } from "enzyme"
import React from 'react'
import '@testing-library/jest-dom';

import {GifGrid} from '../../components/GifGrid'
import { useFechGifs } from "../../hooks/useFechGifs";
jest.mock("../../hooks/useFechGifs");


describe('Prueba sobre <GifGrid />', () =>{
    const category = "Batman";
//    const wrapper = shallow(<GifGrid category = {category}/>); 

 /*   beforeEach(() => {
        jest.clearAllMocks();
        wrapper = shallow(<GifGrid category = {category}/>);
    });
   */ 
    test('debe mostrar un <GifGrid /> correctamente', () => {
        useFechGifs.mockReturnValue({
            data: [],
            loading: true
        })

        const wrapper = shallow(<GifGrid category = {category}/>);
        expect(wrapper).toMatchSnapshot();
    });

    test('debe mostrar items cuando se cargan imÃ¡genes useFetchGifs', () => {
        const gifs = [{
            id:'ABC',
            title:'algo',
            url:'https://localhost/algo/algo.jpg'
        },
        {
            id:'1234',
            title:'algo',
            url:'https://localhost/algo/algo.jpg'
        }]
        useFechGifs.mockReturnValue({
            data: gifs,
            loading: false
        })

        const wrapper = shallow(<GifGrid category = {category}/>);
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('p').exists()).toBe(false);
        expect(wrapper.find('GifGridItem').length).toBe(gifs.length);
    });
        
})