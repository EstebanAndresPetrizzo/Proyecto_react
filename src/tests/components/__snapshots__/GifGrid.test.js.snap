// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba sobre <GifGrid /> debe mostrar items cuando se cargan im√°genes useFetchGifs 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__zoomIn"
  >
    Batman
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      title="algo"
      url="https://localhost/algo/algo.jpg"
    />
    <GifGridItem
      id="1234"
      key="1234"
      title="algo"
      url="https://localhost/algo/algo.jpg"
    />
  </div>
</Fragment>
`;

exports[`Prueba sobre <GifGrid /> debe mostrar un <GifGrid /> correctamente 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__zoomIn"
  >
    Batman
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
